include/alpaka/mem/buf/BufUniformCudaHipRt.hpp

```diff
@@ -89,21 +85,8 @@ namespace alpaka
                 "The idx type of TExtent and the TIdx template parameter have to be identical!");
         }
 
-    private:
-        //! Frees the shared buffer.
-        ALPAKA_FN_HOST static auto freeBuffer(TElem* const memPtr, DevUniformCudaHipRt const& dev) -> void
-        {
-            ALPAKA_DEBUG_MINIMAL_LOG_SCOPE;
-
-            // Set the current device.
-            ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(ALPAKA_API_PREFIX(SetDevice)(dev.m_iDevice));
-            // Free the buffer.
-            ALPAKA_UNIFORM_CUDA_HIP_RT_CHECK(ALPAKA_API_PREFIX(Free)(reinterpret_cast<void*>(memPtr)));
-        }
-
     public:
-        DevUniformCudaHipRt m_dev; // NOTE: The device has to be destructed after the memory pointer because it is
-                                   // required for destruction.
```

Comment:
The comment is no longer relevant, because the deleter captures the device by value.